src/rust/engine/process_executor/src/remote.rs

```diff
@@ -0,0 +1,74 @@
+use std::error::Error;
+
+use bazel_protos;
+use crypto::digest::Digest;
+use crypto::sha2::Sha256;
+use protobuf;
+
+use super::{ExecuteProcessRequest, ExecuteProcessResult};
+
+///
+/// Runs a command via a gRPC service implementing the Bazel Remote Execution API
+/// (https://docs.google.com/document/d/1AaGk7fOPByEvpAbqeXIyE8HX_A3_axxNnvroblTZ_6s/edit).
+///
+pub fn run_command_remote(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+  let mut command = bazel_protos::remote_execution::Command::new();
+  command.set_arguments(protobuf::RepeatedField::from_vec(req.argv));
+  for (name, value) in req.env {
+    let mut env =  bazel_protos::remote_execution::Command_EnvironmentVariable::new();
+    env.set_name(name);
+    env.set_value(value);
+    command.mut_environment_variables().push(env);
+  }
+
+  let mut action = bazel_protos::remote_execution::Action::new();
+  action.set_command_digest(digest(&command)?);
```

Comment:
Sidenote: it's interesting that this requires double-serializing things for the purposes of digesting them... perhaps there are "set_field_as_raw_bytes" type APIs on the Java/C++ side to avoid double serializing? *shrug

Reply:
Yeah, a little annoying. I guess we can try to hack something clever into place if this becomes noticeable :)